pkg/Cpanel/Security/Advisor/Assessors/Permissions.pm

```diff
@@ -55,7 +55,11 @@ sub _check_for_unsafe_permissions {
             }
         }
         if ( !$perms_ok ) {
-            my $expected_mode = sprintf( "%04o", $expected_attributes->{'perms'} );
+            my $expected_mode = '';
+            foreach my $perms ( @{ $expected_attributes->{'perms'} } ) {
+                $expected_mode .= sprintf( "%04o ", $perms );
```

Comment:
Sure, I don't see why not.  I just hadn't thought about it at the time.

    --Tim


On Mon, Jun 13, 2016 at 9:55 AM, brian m. carlson <notifications@github.com>
wrote:

> In pkg/Cpanel/Security/Advisor/Assessors/Permissions.pm
> <https://github.com/CpanelInc/addon_securityadvisor/pull/38#discussion_r66806215>
> :
>
> > @@ -55,7 +55,11 @@ sub _check_for_unsafe_permissions {
> >              }
> >          }
> >          if ( !$perms_ok ) {
> > -            my $expected_mode = sprintf( "%04o", $expected_attributes->{'perms'} );
> > +            my $expected_mode = '';
> > +            foreach my $perms ( @{ $expected_attributes->{'perms'} } ) {
> > +                $expected_mode .= sprintf( "%04o ", $perms );
>
> Couldn't we use a map and join here, like this?
>
> my $expected_mode = join(' ', map { sprintf('%04o', $_) } @{$expected_attributes->{'perms'}});
>
> That would avoid the need for the cleanup on the next line and would be
> marginally more efficient.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/CpanelInc/addon_securityadvisor/pull/38/files/45bfb3a2ab92a12f627a1b7ca30190ebace5b3d9#r66806215>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABBCZe6DkFSB9LhSeMPpHm0GTxH9brkvks5qLW95gaJpZM4IzanN>
> .
>